core/src/cluster_nodes.rs

```diff
@@ -313,6 +317,19 @@ impl ClusterNodes<RetransmitStage> {
     }
 }
 
+impl ClusterNodes<FindPacketSenderStakeStage> {
```

Comment:
@carllin [suggested](https://github.com/solana-labs/solana/pull/23257/commits/5900f84ac06496625f77efd2a6152b6942ef50af#r819283104) to utilize `ClusterNodesCache`, add this function to return sender (identified by its IP) stakes.

Reply:
I would rather we don't use `ClusterNodesCache`. This cache is meant to implement very turbine specific logic and heavily optimize for turbine use cases; and we will have to keep changing it as we change turbine design. Having other dependencies will tie our hands on what gymnastics we can perform here.

It seems to me for QoS use cases a much simpler implementation will do.

I feel like this is one of those cases that the trade-off leans more towards copy-pasting whatever pieces you need from this code, rather than code-reuse. Though I agree that the later generally tends to be a better approach.

Reply:
Thanks for chiming in, and explaining the intended usage of `ClusterNodesCache`. With that, I agree that re-use isn't proper in this case, at least not at the moment. 

So if we'd copy and simplify `ClusterNodesCache` for this particular purpose, then it's original implementation at [here](https://github.com/solana-labs/solana/pull/23257/commits/5900f84ac06496625f77efd2a6152b6942ef50af#r819283104) should suffice. `clusterNodesCache` calls `bank.epoch_staked_nodes()` and `cluster_info.tvu_peers()`, with TTL; which is pretty much the original implementation is. @carllin wdyt if I just revert it back?

Reply:
@taozhu-chicago yup that sounds good to me. Sorry for the hassle

One thing of note is I think we can refresh less than once every second, a constant `STAKES_REFRESH_PERIOD_IN_MS` around 5 seconds seems in line with the other place where this is cached.

Reply:
Awesome, thank you both for ideas and inputs. Good opportunity to explore. 😄 

Yep, good idea to set refresh rate to 5 sec.